changes.go

```diff
@@ -22,54 +22,294 @@ func FromData(data *charm.BundleData) []*Change {
 }
 
 // Change holds a single change required to deploy a bundle.
-type Change struct {
-	// Id is the unique identifier for this change.
-	Id string `json:"id"`
-	// Method is the action to be performed to apply this change.
-	Method string `json:"method"`
-	// Args holds a list of arguments to pass to the method.
-	Args []interface{} `json:"args"`
-	// Requires holds a list of dependencies for this change. Each dependency
+type Change interface {
+	// Id returns the unique identifier for this change.
+	Id() string
+	// Requires returns a list of dependencies for this change. Each dependency
 	// is represented by the corresponding change id, and must be applied
 	// before this change is applied.
-	Requires []string `json:"requires"`
+	Requires() []string
+	// Method returns the action to be performed to apply this change.
+	Method() string
+	// GUIArgs returns positional arguments to pass to the method, suitable for
+	// being serialized and sent to the Juju GUI.
+	GUIArgs() []interface{}
+	// setId is used to set the identifier for the change.
+	setId(string)
 }
 
-// changeset holds the list of changes returned by FromData.
-type changeset struct {
-	changes []*Change
+type changeInfo struct {
```

Comment:
// changeInfo holds information on a change, suitable
// for embedding into a more specific change type.

?